lib/iris/_shapefiles.py

```diff
@@ -0,0 +1,221 @@
+
+        if xmod:
+            x0, x1 = _rebase_values_to_modulus((x0, x1), xmod)
+        if ymod:
+            y0, y1 = _rebase_values_to_modulus((y0, y1), ymod)
+
+        # create a new polygon of the grid cell and check intersection
+        cell_box = sgeom.box(x0, y0, x1, y1)
+        intersect_bool = trans_geo.intersects(cell_box)
+        # mask all points without a intersection
+        if intersect_bool is False:
+            mask_template[idx] = True
+        # if weights method used, mask intersections below required weight
+        if intersect_bool is True and weights is True:
+            intersect_area = trans_geo.intersection(cell_box).area
+            if (intersect_area / cell_box.area) <= minimum_weight:
+                mask_template[idx] = True
+
+    return mask_template
+
+
+def _transform_coord_system(geometry, cube, geometry_system=None):
+    """Project the shape onto another coordinate system.
+
+    Arguments:
+        target: The target :class:`iris.coord_systems.CoordSystem`
+                or a :class:`iris.cube.Cube` object defining the coordinate
+                system to which the shape should be transformed
+
+    Returns:
+        A transformed shape (copy)
+    """
+    y_name, x_name = _cube_primary_xy_coord_names(cube)
+    DEFAULT_CS = _get_global_cs()
+    target_system = cube.coord_system()
+    if not target_system:
+        warnings.warn(
+            "Cube has no coord_system; using default GeogCS lat/lon",
+            IrisUserWarning,
+        )
+        target_system = DEFAULT_CS
+    if geometry_system is None:
+        geometry_system = DEFAULT_CS
+    target_proj = target_system.as_cartopy_projection()
+    source_crs = geometry_system.as_cartopy_projection()
+
+    trans_geometry = target_proj.project_geometry(geometry, source_crs)
+    if target_system == DEFAULT_CS and cube.coord(x_name).points[-1] > 180:
+        trans_geometry = shapely.transform(trans_geometry, _trans_func)
+
+    return trans_geometry
+
+
+def _trans_func(geometry):
+    """pocket function for transforming the x coord of a geometry from -180 to 180 to 0-360"""
+    for point in geometry:
+        if point[0] < 0:
+            point[0] = 360 - np.abs(point[0])
+    return geometry
+
+
+def _get_global_cs():
+    """pocket function for returning the iris default coord system to avoid circular imports"""
+    import iris.fileformats.pp
+
+    DEFAULT_CS = iris.coord_systems.GeogCS(iris.fileformats.pp.EARTH_RADIUS)
+    return DEFAULT_CS
+
+
+def _cube_primary_xy_coord_names(cube):
+    """Return the primary latitude and longitude coordinate standard names, or
+    long names, from a cube.
+
+    Arguments:
+        cube (:class:`iris.cube.Cube`): An Iris cube
+
+    Returns:
+        The names of the primary latitude and longitude coordinates
+    """
+    latc = cube.coords(axis="y")[0] if cube.coords(axis="y") else -1
+    lonc = cube.coords(axis="x")[0] if cube.coords(axis="x") else -1
```

Comment:
Given this is ran 4 times total, not a optimisation priority I think

Reply:
See how this one plays out: https://github.com/SciTools/iris/pull/5470#discussion_r1400698757

Because I should have mentioned this isn't just a performance thing but consistency with other internal Iris code.

Reply:
the "axis" keyword for .coords doesn't accept more than one input according to the documentation. I've put in a dim_coords check because discussion with patrick indicated that this code will fail with 2D x/y coordinates, but the code doesn't explicitly forbid them.